### PR TITLE
FIX: Always preload admin plugin list for admin in sidebar

### PR DESCRIPTION
**Pull request by martin-brennan**
When we show the links to installed plugins in the admin
sidebar (for plugins that have custom admin routes) we were
previously only doing this if you opened /admin, not if you
navigated there from the main forum. We should just always
preload this data if the user is admin.

This commit also changes `admin_sidebar_enabled_groups` to
not be sent to the client as part of ongoing efforts to
not check groups on the client, since not all a user's groups
may be serialized.

### CODE REVIEW

File: spec/requests/application_controller_spec.rb [hunk 1]
```diff
@@ -1308,6 +1308,7 @@ def preloaded_json
             "topicTrackingStates",
             "topicTrackingStateMeta",
             "fontMap",
+            "enabledPluginAdminRoutes",
```
Comment (tgxworld) [inline spec/requests/application_controller_spec.rb:1311]: This is just testing that the key exists but I think it is important to test the content as well.

File: app/serializers/current_user_serializer.rb [hunk 2]
```diff
@@ -122,6 +124,10 @@ def can_send_private_messages
     scope.can_send_private_messages?
   end
 
+  def use_admin_sidebar
+    object.admin? && object.in_any_groups?(SiteSetting.admin_sidebar_enabled_groups_map)
+  end
```
Comment (tgxworld) [inline app/serializers/current_user_serializer.rb:129]:
```suggestion
  def use_admin_sidebar
    object.in_any_groups?(SiteSetting.admin_sidebar_enabled_groups_map)
  end

  def include_user_admin_sidebar?
    object.admin?
  end
```

Minor but there is no need to include the attribute for a user that is not an admin.

**Review by tgxworld — APPROVED** [review]
Some comments but nothing blocking here.

File: app/serializers/current_user_serializer.rb (hunk 2 above)
Comment (martin-brennan) [reply app/serializers/current_user_serializer.rb:129]: Ah yeah good point!

**Pull request merged**